src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Presentation/CompletionPresenterSession.cs

```diff
@@ -165,6 +179,12 @@ public void Dismiss()
         {
             AssertIsForeground();
 
+            // if completion is not shown to users yet, then
```

Comment:
This comment doesn't jive with this method: this is called if the list is shown and then dismissed by user action, right? What's the intent here?

Reply:
@Pilchie @rchande wants time between the point we start processing completion set and the point UI is shown to users.

this can't be added to editor since we dont follow editor's API and dont start session until we ask editor to show the UI, we also can't be added since we don't know when editor actually show UI (editor don't show UI even if we asked them. they decide when to show once we ask them)

so, after talking with editor team, whether they add new API to let us know when UI is painted or new API we can call to mark when session is started..

decided to just re-use BestMatch. according to them, it is first API they call once UI is actually painted.

if you are talking with them on new API, you should talk about this need as well. ask @Pilchie since he was involved in the discussion as well.

...

about why this particular code is here is that, we need to distinguish cases that completion is dismissed before UI is shown and completion is shown and then dismissed. whoever called first will fire the AI event with right cancellation.

Reply:
Oh I see, we're assuming that if ReportPerformance() got called, at that point this invoking of the cancellation won't do anything. Got it....can we make the comment clearer?

Reply:
sure.